### PR TITLE
Prefer using plugin extensions over deprecated conventions

### PR DESCRIPTION
**Pull request by Goooler**
https://docs.gradle.org/current/userguide/upgrading_version_7.html#all_convention_deprecation

### CODE REVIEW

**Comment by Goooler** [comment]
CC @nedtwigg

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java [hunk 1]
```diff
@@ -26,13 +26,14 @@
 
 public class SpotlessPlugin implements Plugin<Project> {
 	static final String SPOTLESS_MODERN = "spotlessModern";
-	static final String MINIMUM_GRADLE = "6.1.1";
+	static final String VER_GRADLE_min = "6.1.1";
+	static final String VER_GRADLE_javaPluginExtension = "7.1";
```
Comment (Goooler) [inline plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java:30]: May need to bump `VER_GRADLE_min` to `7.1`, using conventions will start emitting deprecation warnings in Gradle 8.1, and will be removed in the future Gradle 8 versions.

Comment (nedtwigg) [reply plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java:30]: Roger. If we are throwing warnings that we can't easily workaround with a version check, then I'm happy to bump the minimum.

**Comment by nedtwigg** [comment]
Thanks!

**Comment by nedtwigg** [comment]
Published in `plugin-gradle 6.18.0`.

File: plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java (hunk 1 above)
Comment (Goooler) [reply plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java:30]: @nedtwigg Are you planning to go on Gradle plugin 7? If so can we bump the ming Gradle requirement to 7.1 for something like this?

Comment (Goooler) [reply plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessPlugin.java:30]: And 682dcd6d4840002adb834b2db67038869efe5d57.

**Pull request merged**